src/main.rs

```diff
@@ -76,17 +77,31 @@ fn main() {
         .and_then(|dopt| dopt.deserialize())
         .unwrap_or_else(|e| e.exit());
     set_threading(args.flag_threads);
+
     let mut header = None;
+    let mut ccache = CountCache::new();
+    for elem in args.arg_elems.iter() {
+        match elem.parse() {
+            Ok(Formula::Image(image_path)) => {
+                if ccache.has_key(image_path.clone()) {
```

Comment:
I'm not gonna comment too mucu on this `CountCache` thing, but you should probably make it respect the [HashMap ](https://doc.rust-lang.org/std/collections/struct.HashMap.html) definition, like `contains_key` instead of `has_key`, which should receive a `&T` instead of `T`, etc.